src/Features/CSharp/Portable/UseLocalFunction/CSharpUseLocalFunctionCodeFixProvider.cs

```diff
@@ -148,13 +148,15 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
                 ? new SyntaxTokenList(anonymousFunction.AsyncKeyword)
                 : default;
 
-            var returnType = delegateMethod.GenerateReturnTypeSyntax();
-
             // We add a space directly after the return-type, as the default presence of an elastic 
             // trivia makes the formatting engine thing it can take a single-line local function and
             // wrap it over multiple lines.
-            var returnType = invokeMethod.GenerateReturnTypeSyntax().WithTrailingTrivia(SyntaxFactory.Space);
-            
+            var returnType = delegateMethod.GenerateReturnTypeSyntax().WithTrailingTrivia(SyntaxFactory.Space);
+            if (modifiers.Count > 0) // if it's async, we need to add one before the return type as well...
```

Comment:
@CyrusNajmabadi's fix didn't work for async

Reply:
should we put the space on the trailing trivia on the async keyword instead?

Reply:
alternatively, we could create the entire thing, then do a final:  localFunction.ReplaceToken(localFunction.GetFirstToken(), localFunction.GetFirstToken().WithTrailingTrivia(...))

Reply:
> should we put the space on the trailing trivia on the async keyword instead?
I tried that, it didn't work

Reply:
> localFunction.ReplaceToken(localFunction.GetFirstToken(), localFunction.GetFirstToken().WithTrailingTrivia(...))

didn't work either. and if you had a more complex type, a space after the first token inside it might be undesirable